src/controllers/session.ts

```diff
@@ -0,0 +1,60 @@
+import { Response, NextFunction } from 'express';
+import { RequestWithUserRole, AppointmentWithTherapistOptional } from '../types';
+import generateMeeting from '../services/calendar';
+import { templateErrors } from '../helpers';
+import { BookedAppointment } from '../services';
+import { User } from '../models';
+
+const createSessionController = async (
+  req: RequestWithUserRole,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    const userData = req.user;
+    const { appointmentId } = req.body;
+
+    const appointment: AppointmentWithTherapistOptional
+    | null = await BookedAppointment(appointmentId);
+
+    if (!appointment) {
+      throw templateErrors.BAD_REQUEST('The requested appointment is not available.');
+    }
+    if (!userData) {
+      throw templateErrors.UNAUTHORIZED('user not found');
+    }
+
+    appointment.isBooked = true;
+    await appointment.save();
+
+    const therapistEmail = appointment?.therapist?.user?.email;
+    if (!therapistEmail) {
+      throw templateErrors.BAD_REQUEST('The Therapist email is not available');
+    }
+    const user = await User.findOne({
+      attributes: ['email'],
+      where: { id: userData.userId },
+    });
+
+    const userEmail = user?.email;
+    if (!userEmail) {
+      throw templateErrors.BAD_REQUEST('The Therapist email is not available');
```

Comment:
```suggestion
      throw templateErrors.BAD_REQUEST('Something went wrong,The user email is not available);
```
Same here it's not a bad request that we can't find user's email

Reply:
we can create custom error with 500 status